### ISSUE TITLE
Trans component does not display children

### ISSUE THREAD

**Issue by multiwebinc** [i18next/i18next#1582]
Here is a rather contrived example, but say you have the following:

```jsx
<Trans
  i18nKey="button"
  components={{
    hello: (
      <button>Hello</button>
    )
  }}
/>
```

And in the translation file you have:

```json
{ "button": "Please say <hello />" }
```

This produces the following HTML:

```html
Please say <button></button>
```

The children of that component are not rendered.

However, if I use another component, then it does display the children:

```jsx
<Trans
  i18nKey="button"
  components={{
    hello: <GetButton />
  }}
/>
```

```
const GetButton = () => {
  return <button>Hello</button>
}
```

I'm not sure if this is by design or what, but I think it should output the child elements as well, not just the parent.

**Comment by adrai** [comment]
Try with:

{ "button": "Please say <hello>Hello!!!</hello>" }

**Comment by multiwebinc** [comment]
@adrai, Yes, that works, but as I said, the example was contrived, but maybe I made it too simple. The way you are doing it is limited to strings, but putting other components or JSX in there does not work. Say for example, I have this:

```jsx
<Trans
  i18nKey="guest_form.terms_label"
  components={{
    bookingAndCancellationPolicies: (
      <span className="label-fake-link">
        <PolicyModal
          text={i18n.t('guest_form.booking_policy')}
        >
          <BookingPolicySummary />
        </PolicyModal>
      </span>
    )
  }}
/>
```

And in my translation file I do:

```
"terms_label": "I acknowledge that I have read and accept the <bookingAndCancellationPolicies />.",
```

Where `<PolicyModal />` is a reusable component that brings up a modal window that contains the logic to fetch the policy from an external source.

**Comment by jamuhl** [comment]
PR welcome...

Your sample could be solved as:

```js
const MyComp = () => (<span className="label-fake-link">
        <PolicyModal
          text={i18n.t('guest_form.booking_policy')}
        >
          <BookingPolicySummary />
        </PolicyModal>
      </span>);
      
      
<Trans
  i18nKey="guest_form.terms_label"
  components={{
    bookingAndCancellationPolicies: MyComp
  }}
/>
```

**Comment by multiwebinc** [comment]
@jamuhl Yes, that's precisely what I am currently doing.

I tried to see if I could make a PR, but I came upon some inexplicable behavior in my browser that makes debugging what's happening nearly impossible. Somewhere along the line, the children are being stripped off, but I can't figure out where that is.

In Trans.js, line 165 is

```js
const child =
          Object.keys(node.attrs).length !== 0 ? mergeProps({ props: node.attrs }, tmp) : tmp;
```

if I insert the following after that:

```js
console.log('obj', child)
console.log("obj.children", child.children)
```
I get different values for the children in each case, which makes no sense to me.

![image](https://user-images.githubusercontent.com/901732/112226541-e332c080-8bf3-11eb-9cd8-28e892ae8f3d.png)

In the first line, `child.children` is an array, and in the second line it is an object. 

In the first line, `child.children[0].props.children` is an empty array


, and in the second line, `child.children.hello.props.children` contains the child elements for that node. I'm totally confused by this.
